pkg/trtl/namespaces.go

```diff
@@ -0,0 +1,9 @@
+package trtl
+
+// Reserved namespaces that cannot be used by the caller since they are in use by trtl.
+// If necessary this can be moved to configuration in the future.
+var reservedNamespaces = map[string]struct{}{
+	"peers":    struct{}{},
+	"sequence": struct{}{},
+	"index":    struct{}{},
+}
```

Comment:
Is it a common go pattern to populate value-less maps with anonymous structs or are we just leaving room for future namespace configuration?

Reply:
Great question, this is an idiomatic and lightweight mechanism to [store a set in Go](https://golangbyexample.com/set-implementation-in-golang/). The empty anonymous structs don't allocate any memory unlike a `bool` which allocates a bit.

Reply:
Cool!